R/sample.R

```diff
@@ -0,0 +1,89 @@
+utils::globalVariables(c("Assay_Plate_Barcode", "replicate_id"))
+#' Sample selects replicates across specified plates.
+#'
+#' \code{sample} replicates across specified plates, and aggregates the result in a single file.
+#'
+#' @param batch_id        Batch ID.
+#' @param pattern         Regular expression - only CSVs.
+#' @param output          Output file - either CSV or RDS.
+#' @param replicates      Number of replicates to select per plate map. Selects all replicates if \code{NULL}. default: \code{NULL}.
+#' @param workspace_dir   Root directory containing backend and metadata subdirectories. Can be relative or absolute. default: \code{"."}.
+#' @importFrom magrittr %<>%
+#' @importFrom magrittr %>%
+#' @importFrom magrittr extract2
+#' @importFrom rlang .data
+#' @importFrom utils head
+#' @importFrom utils tail
+#' @export
+sample <- function(batch_id, pattern, output,
+                   replicates = NULL,
+                   workspace_dir = ".") {
+  backend_dir <- paste(workspace_dir, "backend", batch_id, sep = "/")
```

Comment:
instead of paste(workspace_dir, "backend", batch_id, sep = "/") use file.path(workspace_dir, "backend", batch_id)